### ISSUE TITLE
Drony: do not work with socks5 wall breaker proxy while Firefox mobile can

### ISSUE THREAD

**Issue by luweitest** [SuppSandroB/sandrop#134]
I have a GFW breaker proxy in computer: 192.168.1.109:8080 (http) or 1080(socks5)
On a non-root android phone 192.168.1.100, in Firefox mobile, set proxy using about:config and succeed.
Then in Drony using normal settings, GFW is not broken.
But Drony works with normal proxy. I suspect that it may be related to remote DNS. Drony do not have an option like network.proxy.socks_remote_dns in Firefox mobile.

**Comment by SuppSandroB** [comment]
Drony has options for dns. For example you can use dnscrypt. Or you can specify where dns requests are send with tcp/ip address.

**Comment by luweitest** [comment]
On 2018-4-7 23:29, SuppSandroB wrote:
> Drony has options for dns. For example you can use dnscrypt.

Tried dnscrypt with no success.

> Or you can specify where dns requests are send with tcp/ip address.

Is it "DNS cache settings"? The default listening port of it is 8054
instead of 53, so I thought it is not normal DNS setting.

What is the equivalent setting to remote DNS with socks5 proxy?

And in the situation of http proxy, normally dns is not required because
the whole http request is send to the proxy by browser. Does Drony work
like that?

-- 
Regards,
Lu Wei
PGP key ID: 0x A12F EF75 92CC E1EA

**Comment by luweitest** [comment]
Setting "DNS cache settings" to 8.8.8.8:53, not work.
Double checked plain HTTP, it now works. So only Socks5 do not work.

**Comment by brookhong** [comment]
Same issue. Drony works well with plain proxy, but not with socks5.

**Comment by SuppSandroB** [comment]
will look at it

**Comment by SuppSandroB** [comment]
Drony should work with SOCK5 proxy. But not all features of SOCKS5 are supported.
Can you send example of configuration that is not working?

**Comment by mut4nt** [comment]
Hello,

Thank you for your great job on drony.

I have the same exact problem and I also suspected remote dns option. I also tried on firefox mobile first as it is the common way to test socks5 from its official website. Just the same exact thought process.

Seems the remote dns is the issue because when the socks5 proxy receive request from another client (like Firefox) no Ip address is directly involved. The proxy will do the dns job himself. But when the request come from drony ip address is shown. That means drony resolves the dns locally/himself. I just suggest to add an option just as Firefox does : "remote dns" that let the socks5 proxy do it itself.

Will put some time on it if possible because I really want this feature. My proxy has trouble handling ip address directly.

Best regards,

**Comment by brookhong** [comment]
@SuppSandroB 

I agree with @mut4nt, it looks like drony always try to resolve DNS locally by itself. But my socks5 proxy is set up with `ssh`, and it can only be used as

    curl --socks5-hostname <proxy_ip>:1080 https://www.google.com
    
it does not work with

    curl --socks5 <proxy_ip>:1080 https://www.google.com
